projects/RabbitMQ.Client/RabbitMQ.Client.csproj

```diff
@@ -75,12 +75,12 @@
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="all" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="8.0.0" PrivateAssets="all" />
     <PackageReference Include="MinVer" Version="4.3.0" PrivateAssets="all" />
-    <PackageReference Include="System.Memory" Version="4.5.5" />
-    <PackageReference Include="System.Threading.Channels" Version="8.0.0" />
     <PackageReference Include="System.IO.Pipelines" Version="8.0.0" />
   </ItemGroup>
 
   <ItemGroup Condition="$(TargetFramework) == 'netstandard2.0'">
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.2" />
```

Comment:
Should this be updated to `8.0.0` also

Reply:
In general, dependencies should be the lowest possible, to not constrain users that use multiple libraries.

Reply:
Agree we should not bump dependencies unless we need new functionality or the current version has CVE etc.

Reply:
I subscribe to the school of thinking that, unless a dependency is known to be fragmented and is really essential (e.g. library X is extremely popular and known to have different majors to be in used at the same time, for any reason), we should keep all of them up-to-date.

We don't want our users to be years behind now, do we? Even though I get that it is somewhat different for libraries.

But it's probably not particularly essential to bump those two immediately.

Reply:
> We don't want our users to be years behind now, do we? Even though I get that it is somewhat different for libraries.

By depending on lower versions, you're not forcing users to be behind.

By depending on higher versions, you might be forcing users to use lower versions of your library.

Reply:
Which is why above I mention the fact that some dependencies are more popular than others. For less popular ones, it is unlikely that you are going to affect anyone's dependencies.

Anyhow, in several programming language communities "bump early and often" is largely the norm. If that's not the case with modern .NET, I'm not going to argue with that :)

Reply:
it's ok to me from either side, 
then I have a question, why did we upgrade to `8.0.0` for `System.Threading.Channels`, seemed there's no diff between `6.0.0`/`7.0.0` and `8.0.0`

https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1347/files#diff-578fb561d8fb661e0bf3e00dd4a9efeb4c7ad9d6654ab26b8a7aeca19732f6aaR73

Since `7.0.0` is a major change, think it may be acceptable to use the latest stable dependency, especially for some "Core" dependencies for performance improvements/bug fix

Reply:
I think it is important to point out those packages are only for users using a version of dotnet older than net 6 but compliant with net standard 2.0 (net 4.6.2 I think). At which point we should be encouraging users to update their framework to a new version as that way it will always be kept up to date.

Reply:
I'm afraid we can't, really: those who are still on .NET 4.6.2 will likely stay there forever (or at least for a long time). Maybe MSFT can nudge them to upgrade somehow (by dropping support for 4.6.2, LOL) but small library authors? Heck, Spring has to still support ancient Java versions, and they have **enormous** weight in that community.

Reply:
https://learn.microsoft.com/en-US/lifecycle/faq/dotnet-framework

.NET Framework is a Windows component and each version is supported until the end of support of the Windows versions that natively have that version.

I don't expect versions 4.7.2 and 4.8.x to be out of support any time soon.

The lowest supported version is 4.6.2 which is fully compliant with .NET Standard 2.0.

But, as .NET diverges from .NET Framework each new version, it become impractical to libraries keep .NET Framework and .NET support without different code bases and features.

Reply:
@WeihanLi 

> then I have a question, why did we upgrade to 8.0.0 for System.Threading.Channels, seemed there's no diff between 6.0.0/7.0.0 and 8.0.0

Because `dotnet outdated -u` upgraded the dependency.

Because of this discussion, I have "downgraded" some of the dependencies in this PR.